topics/RDMA.md

```diff
@@ -0,0 +1,170 @@
+---
+title: "RDMA support"
+linkTitle: "RDMA support"
+description: Valkey Over RDMA support
+---
+
+Valkey supports the Remote Direct Memory Access (RDMA) connection type via a
+Valkey module that can be dynamically loaded on demand.
+
+## Getting Started
+
+RDMA enables direct data exchange between networked computers' main memory,
+bypassing processors and operating systems.
+
+As a result, RDMA offers better performance compared to TCP. Test results indicate that
+Valkey Over RDMA achieves approximately 2 times higher QPS and lower latency.
+
+Please note that Valkey Over RDMA is currently supported only on Linux.
+
+## Running manually
+
+To run a Valkey server with RDMA mode:
+
+    ./src/valkey-server --protected-mode no \
+         --loadmodule src/valkey-rdma.so bind=192.168.122.100 port=6379
+
+Bind address/port of RDMA can be modified at runtime using the following command:
+
+    192.168.122.100:6379> CONFIG SET rdma-port 6380
+
+Valkey can run both RDMA and TCP concurrently on the same port:
+
+    ./src/valkey-server --protected-mode no \
+         --loadmodule src/valkey-rdma.so bind=192.168.122.100 port=6379 \
+         --port 6379
+
+Note that the network interface (192.168.122.100 of this example) should support
+RDMA. To test a server supports RDMA or not:
+
+    ~# rdma dev show (a new version iproute2 package)
+Or:
+
+    ~# ibv_devices (ibverbs-utils package of Debian/Ubuntu)
+
+
+## Protocol
```

Comment:
I found this section really confusing, and had to read through it a couple of times to really understand what it was trying to say. If the intention is for a client to implement the protocol, it seems like they need a good bit of understanding about RDMA to do so. If we're just trying to give an overview, I wonder how much is useful here vs just keeping it in .c file.

Like, a concrete point is what is the point of the 4 new opcodes we are introducing. Are they part of the RDMA spec or are they something special we are adding. Since this topic should stand on it's own for people who are finding it on the website or man page, I think it needs to make it clearer what its purpose is.

Reply:
>If the intention is for a client to implement the protocol

That is a good point. I was under the impression that this is meant to be an "introduction" that shows the high level idea without getting into too much low level detail. But going through @madolson's comments, I realize that there is a lot more value in defining a real RFC spec that can be used to guide the client implementation than a high level overview. We should pivot.

Reply:
I'm trying to say:
* the Valkey Over RDMA protocol(`Protocol` section). Base on this specification, we can implement both the server side, and the client sides in different programming language. Just like `valkey-doc-pizhenwei/topics/protocol.md`
* how to run Valkey Over RDMA(`Getting Started`).

Do you prefer to separate this doc into two? or any suggestions?

Reply:
Viewing this as a specification that can be implemented by a client, it seems that it would be useful to include a link to the code for any client(s) that already exist.  Perhaps one could be given the status of "reference implementation".